### ISSUE TITLE
uint64_t typemap to Java altered from correct long to BigInteger after SWIG upgrade to 4.2.1 (with SWIGWORDSIZE64)

### ISSUE THREAD

**Issue by blauc** [swig/swig#2947]
With SWIG versions < 4.2 , unsigned fixed-size 64 bit integer type `uint64_t` were correctly mapped to Java type `long`.

Now, these become wrapped as java type BigInteger, which I find undesireable, because it is not a primitive datatype, breaks backwards compability and generates a sizable overhead over the primitive datatype.

The new mapping occurs due to changes in `stdint.i` in combination with `typedefs.i`.

```
stdint.i
20 : #if defined(SWIGWORDSIZE64)
21 : typedef unsigned long int	uint64_t;
22 : #else

typemaps.i

257: OUTPUT_TYPEMAP(unsigned long long, jobject, java.math.BigInteger, Object, "[Ljava/math/BigInteger;", jobjectArray);
...
545: %apply unsigned long long *OUTPUT { unsigned long *OUTPUT };
etc
...

```
To see this behaviour, use SWIGWORDSIZE64.

**Comment by wsfulton** [comment]
Can you please provide an example of what you are using? You mention typemaps.i, but there is no support for `uint64_t` in typemaps.i and SWIG has always generated a Java BigInteger for C type `uint64_t` when using stdint.i, so I don't understand what has changed if you are simply using `uint64_t`. Maybe you are applying some typemaps? If so, you probably need to change the types you are applying.

I'm guessing , but the change might be due to 7bba06b03e.

**Comment by blauc** [comment]
You are right - what happend was that previous code previously mapped `uint64_t` to `long` and change https://github.com/swig/swig/commit/7bba06b03ef4529c542c3cf8364c382027aae000 altered this behaviour indirectly. So nothing to fix from the SWIG side. Thanks for commenting and helping out!
